app/modules/collaborations/models.py

```diff
@@ -17,13 +51,126 @@ class Collaboration(db.Model, HoustonModel):
     )  # pylint: disable=invalid-name
-    title = db.Column(db.String(length=50), nullable=False)
+    title = db.Column(db.String(length=50), nullable=True)
+
+    collaboration_user_associations = db.relationship(
+        'CollaborationUserAssociations', back_populates='collaboration'
+    )
+
+    def __init__(
+        self, user_guids=None, approval_states=None, is_initiator=None, **kwargs
+    ):
+        # I think this can be adapted to allow >2 User collabs, opening the door for Org-level.
+        if user_guids is None or len(user_guids) != 2:
+            raise ValueError(
+                '__init__ Collaboration: Creating a collaboration requires a user_guids list.'
+            )
+
+        # You can create a collaboration where both users 'approved'
+        if approval_states is not None and len(approval_states) != len(user_guids):
+            raise ValueError(
+                '__init__ Collaboration: Length of approval_states was not equal to provided user list.'
+            )
+
+        if is_initiator is not None and len(is_initiator) != len(user_guids):
+            raise ValueError(
+                '__init__ Collaboration: Length of is_initiator states was not equal to provided user list.'
+            )
+
+        for i in range(len(user_guids)):
+            from app.modules.users.models import User
+
+            user_tup = User.ensure_edm_obj(user_guids[i])
+
+            if user_tup is None or user_tup[0] is None:
+                raise ValueError(
+                    '__init__ Collaboration: One of the user_guids provided was had no associated user.'
+                )
+
+            collab_user_assoc = CollaborationUserAssociations(
+                collaboration=self, user=user_tup[0]
+            )
+
+            if (
+                approval_states is not None
+                and approval_states[i] in CollaborationUserState.ALLOWED_STATES
+            ):
+                collab_user_assoc.read_approval_state = approval_states[i]
+            if is_initiator is not None and isinstance(is_initiator[i], bool):
+                collab_user_assoc.initiator = is_initiator[i]
+                # If you initiate you approve
+                collab_user_assoc.read_approval_state = CollaborationUserState.APPROVED
+
+            with db.session.begin():
+                db.session.add(collab_user_assoc)
+                db.session.merge(user_tup[0])
+
+    def get_users(self):
+        users = []
+        for association in self.collaboration_user_associations:
+            users.append(association.user)
+        return users
+
+    def get_initiators(self):
+        users = []
+        for association in self.collaboration_user_associations:
+            if association.initiator is True:
+                users.append(association.user)
+        return users
+
+    def get_read_state(self):
+        read_state = CollaborationUserState.APPROVED
+        for association in self.collaboration_user_associations:
```

Comment:
Not sure I'm following what this is trying to achieve. 
Two users in a collaboration, one pending, one declined.
The value returned will depend on the order of the users in the list?

Reply:
This method is supposed to be friendly to collaborations with more than two users in the future, but since you raised this I see it has _several_ problems. I'll tear it up and get something better in there.

Reply:
still struggling with this. 
Two associations, one pending one declined. As the previous read_state is not checked, the outcome will be whichever is the second.
Surely the PENDING check should be 
elif association.read_approval_state == CollaborationUserState.PENDING:
    if read_state != CollaborationUserState.DECLINED:
        read_state = CollaborationUserState.PENDING

Reply:
Aha, I see the break, yes this is probably OK but needs a bit more testing to be certain.